### PR TITLE
feat:Add transaction reviews and history UI

### LINKED ISSUE #29: HU-MKT-07-transaction-history
# User can view their transaction history

**Domain:** Marketplace

## User Story

As a authenticated ITESO user, I want to view their transaction history so that a complete history of their past transactions including items sold, donated, exchanged, and received. each entry should show the item, the other party involved, transaction type, date, and final status. users can filter by transaction type and date range..

## Descripción detallada

Users need access to a complete history of their past transactions including items sold, donated, exchanged, and received. Each entry should show the item, the other party involved, transaction type, date, and final status. Users can filter by transaction type and date range. 
This feature is for the web platform. Priority: medium. 
This story focuses on the core functionality. Advanced features, edge cases, and optimizations may be addressed in separate stories.

## Acceptance Criteria

- [ ] User can access the view
- [ ] All relevant information is displayed
- [ ] Data is formatted correctly
- [ ] Loading states are shown while fetching data
- [ ] Error messages are clear and actionable

### Backend
- GET endpoint for resource retrieval
- Query parameters for filtering/pagination
- Response formatting and serialization

### Frontend
- Component for main UI
- Form validation (if applicable)
- Loading and error states
- Success/error notifications
- Responsive design for mobile and desktop
- API integration with backend endpoints

## Testing Notes

- Test with valid data (happy path)
- Test with invalid/missing required data
- Test with edge cases (empty strings, special characters, very long inputs)
- Test error handling and error messages
- Test loading states and async behavior
- Test with authenticated and unauthenticated users

## Test Data Required

- Test user accounts (authenticated ITESO users)
- Sample items with various categories
- Items in different states (published, draft, deleted)
- Valid input data for all required fields
- Invalid input data for validation testing
- Edge case data (boundary values, special characters)

## Potential Mocks

- Database queries for unit tests

## Dependencies & Duplication Check

**Dependencies:** None identified

**Possible duplicates:** None

## Assumptions

- User is authenticated as an ITESO community member
- Backend API follows RESTful conventions
- Frontend uses Next.js and TypeScript
- Database is PostgreSQL

## ⚠️ Warnings

- Acceptance criterion may be too vague: 'Data is formatted correctly'

### LINKED ISSUE #30: HU-MKT-08-rate-transaction
# User can rate and review a completed transaction

**Domain:** Marketplace

## User Story

As a authenticated ITESO user, I want to rate and review a completed transaction so that rate each other (1-5 stars) and leave an optional written review. ratings contribute to a user's reputation score visible on their profile. reviews help build trust within the iteso community..

## Descripción detallada

After a transaction is completed, both parties should be able to rate each other (1-5 stars) and leave an optional written review. Ratings contribute to a user's reputation score visible on their profile. Reviews help build trust within the ITESO community. 
This feature is for the web platform. Priority: medium. 
This story focuses on the core functionality. Advanced features, edge cases, and optimizations may be addressed in separate stories.

## Acceptance Criteria

- [ ] User can access the view
- [ ] All relevant information is displayed
- [ ] Data is formatted correctly
- [ ] Loading states are shown while fetching data
- [ ] Error messages are clear and actionable

### Backend
- GET endpoint for resource retrieval
- Query parameters for filtering/pagination
- Response formatting and serialization

### Frontend
- Component for main UI
- Form validation (if applicable)
- Loading and error states
- Success/error notifications
- Responsive design for mobile and desktop
- API integration with backend endpoints

## Testing Notes

- Test with valid data (happy path)
- Test with invalid/missing required data
- Test with edge cases (empty strings, special characters, very long inputs)
- Test error handling and error messages
- Test loading states and async behavior
- Test with authenticated and unauthenticated users

## Test Data Required

- Test user accounts (authenticated ITESO users)
- Valid input data for all required fields
- Invalid input data for validation testing
- Edge case data (boundary values, special characters)

## Potential Mocks

- Database queries for unit tests

## Dependencies & Duplication Check

**Dependencies:** None identified

**Possible duplicates:** None

## Assumptions

- User is authenticated as an ITESO community member
- Backend API follows RESTful conventions
- Frontend uses Next.js and TypeScript
- Database is PostgreSQL

## ⚠️ Warnings

- Acceptance criterion may be too vague: 'Data is formatted correctly'

### PR DESCRIPTION
**Pull request by LaloCapacha3**
Introduce transaction review support and a transaction history endpoint/UI. Backend: add TransactionReview model with rating/comment, unique constraint (one review per reviewer per transaction) and validators; register serializers and extend TransactionSerializer with TransactionHistorySerializer and review read/create serializers; add /transactions/history/ list action with filtering and /transactions/{id}/review/ POST action to submit reviews. Frontend: add transaction history page, filters panel, list/card components, star rating and review form, hook to fetch paginated history, API helpers for history and submitting reviews, and type updates. Also add .gitattributes and small navbar/urls cleanup. This enables users to view completed transactions, filter them, and submit a single 1-5 star review per transaction.

### CODE REVIEW

File: backend/marketplace/views/transaction.py [hunk 1]
```diff
@@ -131,3 +133,73 @@ def change_status(self, request, pk=None):
...
+        date_to = request.query_params.get("date_to")
+
+        valid_types = [c for c, _ in Transaction.TRANSACTION_TYPE_CHOICES]
+        if tx_type and tx_type in valid_types:
+            qs = qs.filter(transaction_type=tx_type)
+        if date_from:
+            qs = qs.filter(created_at__date__gte=date_from)
+        if date_to:
+            qs = qs.filter(created_at__date__lte=date_to)
+
```
[the remark this answers is not in the document]
Comment (Ferreirafc1133) [reply backend/marketplace/views/transaction.py:186]: opino lo mismo va a mamar

File: frontend/src/components/transactions/TransactionReviewForm.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        rating,
+        ...(comment.trim() && { comment: comment.trim() }),
+      })) as TransactionReview;
+      onSubmitted(review);
+    } catch (err) {
+      setError(err instanceof Error ? err.message : 'Error al enviar la calificacion.');
+    } finally {
+      setIsLoading(false);
+    }
+  }
```
Comment (Ferreirafc1133) [inline frontend/src/components/transactions/TransactionReviewForm.tsx]: lo mismo sin un body o cualquier respuesta que no sea 204 va a mamar

**Review by Ferreirafc1133 — CHANGES_REQUESTED** [review]
